WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -288,4 +291,30 @@ public void UndoBlock(Height blockHeight)
 	{
 		Coins.SwitchToUnconfirmFromBlock(blockHeight);
 	}
+
+	internal void UpdateRelevantTxs(SmartTransaction tx)
+	{
+		var transactions = TransactionStore.GetTransactions().ToList();
+
+		List<SmartTransaction> relevantTxs = new();
+		foreach (var label in tx.Label)
+		{
+			transactions.Where(tx => tx.Label.Contains(label)).ToList().ForEach(transaction => relevantTxs.Add(transaction));
```

Comment:
I don't understand this. 

I received money from Alice. tx1
I received money from Bob. tx2

I create a new transaction where I am sending coins from tx1 and tx2, adding a label: Satoshi => tx3

Now tx1, tx2, tx3 will have the label of Satoshi, Alice, Bob ?

Reply:
No

![image](https://user-images.githubusercontent.com/54551879/154248587-a43a7b91-81e5-404b-924c-e64f0d7b9905.png)

The logic is the same as at L213 : we only merge them if they were used under the same address, as in that case, in the `Wallet.json` file under the `HdPubKey`s, the labels got merged.
If no merge happened there, neither it will happen here.